### ISSUE TITLE
Music view doesn't save zoom level

### ISSUE THREAD

**Issue by uliska** [frescobaldi/frescobaldi#1247]
Reported by a user (and confirmed by me):

> Howdy Frescobaldi users -- after a recent update to v3 and v3.1, the music view now defaults to 100%. It did not use to do this. Everytime I open a score, I now have to set the view to Page or Width, etc. I cannot find a Configuration option to tell Music View my preferred view (Page).
>
> Am I missing something?
>
> Thanks.

**Comment by wbsoft** [comment]
Use Music->Save view settings as default.
In the preferences Tools->Music View you can enable/disable remembering settings per-document.

**Comment by skiasaurus** [comment]
> In the preferences Tools->Music View you can enable/disable remembering settings per-document.

I have that option selected, but when I close Frescobaldi and then reopen the document I was working on, it reverts to the program default.

**Comment by fedelibre** [comment]
I've just verified on version 3.2 and both the preference option and the menu settings don't work.

**Comment by papioara** [comment]
I can in part confirm fedelibre's comment., although his complaint seems to be more about other mujsic view aspects here like page settings (width/height). 

My complaint, though, is not about this. The above-mentioned settings are definitely remembered, but my problem is the width of the right subwindow of Frescobaldi, which on every launch has to be readjusted. The ratio on startup is always about 2/3 editor subwindow and 1/3 music view subwindow. I cannot find any setting or switch or menu entry where my preferred ratio (half and half editor and music view) can be defined, set or saved.

Any advice?

**Comment by fedelibre** [comment]
You are right.
I'm now testing Frescobaldi current master and it seems that the issue described initially (width setting of each document/tab not remembered) has been fixed: if I set a document to *Width* instead of the default *100*, *Width* is retained.

I confirm that the panes (what you call subwindows) width is not retained.  It's what the tooltip calls "its own layout setting":

```
"If checked, every document in the Music View will remember its\n"
"own layout setting, zoom factor, etc. If unchecked, the View will\n"
"not change its settings when a different document is displayed."
```

**Comment by papioara** [comment]
I have the impression that we are not talking about the same thing yet, but with all these different terms (and maybe I use mine in a way prone to misunderstandings) it is not so easy. 

In order to illustrate better what I am talking about, I attach two screenshots. The first one shows the Frescobaldi window in full-screen mode (which I always use for obvious reasons) after **every** startup. The second one shows what I want to have, and what Frescobaldi seems to 'forget' as soon as I close it (and that did **not** happen in earlier versions).

![Arbeitsfläche 1_001](https://user-images.githubusercontent.com/31970041/221715133-c8235ec2-bfa0-4247-ba20-71a3daffbf18.png)
![Arbeitsfläche 1_002](https://user-images.githubusercontent.com/31970041/221715135-04cdd44e-41ab-44ba-9fed-15e180d3ece7.png)

**Comment by fedelibre** [comment]
Yes, I did understand your issue. To summarize, the pane sizes are not retained:

- when moving from one document/tab to another
- when closing and reopening Frescobaldi

Two symptoms of the same problem, I guess.

I cannot remember if it used to work, but if you are sure that it used to work in the past, then maybe something has changed in Qt and Frescobaldi source has not been updated to adapt to this change. Here's a [list of changes per release in Qt5 dev cycle](https://doc.qt.io/qt-5/qt5-intro.html#new-features-by-qt-version).

**Comment by papioara** [comment]
Ah, ok - I was not a 100% sure. Just one little correction: moving from on tab to another (be it in the editor tabs or in the tool tabs like Music View, Documentation browser or Snippets for example, the size is being retained. It's only upon closing and reopening Frescobaldi.

Btw, the Frescobaldi flatpak shows the same behaviour. Maybe I should just be patient and wait for the next Frescobaldi version and see if it's still there...

**Comment by papioara** [comment]
I have done some further research into Frescobaldi on my machine. I had the idea to compare the Frescobaldi behaviour on different computers and Linux flavors in my house. And to be sure we talk about the same feature: yes, I mean the one which can be controlled by Music > Save current view settings.

Candidate one:
Frescobaldi 2.0.11 on Linux Mint 17.3 Rosa
On launch I get a window split half and half, and there is no 'Save current view settings' yet.
When I set it to Full Screen by pressing F11, the half and half split does not change.

Candidate two:
A Laptop with KaOS, a rolling release radically using Qt for the graphical appearance (that's what they say about it)
Frescobaldi 3.2
On launch I also get a 50/50 window which stays like this when pressing F11. 
When in window mode I choose 'Maximise', the Music View pane goes full-screen and can be toggled back by choosing 'Maximise' again.
When I manually change (in window mode) the 50/50 ratio to something else, choose 'Save current view settings' from the menu and quit Frescobaldi, and when I restart it, it stubbornly reverts to 50/50. Tried several times.

Candidate three:
My work machine with Mint 21.1 Cinnamon LTS
In window mode, on launch, I get a 50/50 window as well, and like on the laptop, I can change the ratios to whatever I like (choosing 'Save current view settings' every time afterwards, but on restarting Frescobaldi I always get my 50/50 split back. I do not mind this so far, because this is what I prefer anyway. It's just for the record.
When I choose 'Maximise' from the menu, the Music View pane surprisingly is detached, and a second 'Maximise' does not toggle it back. You can only do this by a mouse click.

And now we actually come to what exactly I was complaining about, and what I have not sufficiently been able to make clear - all my fault:
When I go to full-screen pressing F11, Frescobaldi follows suit, but different from the other machines and versions I have examined, it does _not_ keep the 50/50 ratio, but goes to something like 65/35.
Since I work on a 1920x1200 monitor, the editor window is way too wide, and the Music View pane too narrow. Using Frescobaldi in full-screen mode on a big monitor is very convenient, but it is annoying to have to manually correct the split every time you fire up Frescobaldi. It is certainly not the end of the world, and I can live with it if I have to. Frescobaldi is too good a tool and offers so much Lilypond convenience that I certainly won't stop using it because of this little flaw.

But, of course I wanted to put the question in order to see, if there is something I may have overlooked, so I can change it. Or maybe this is a little bug which could be fixed at some point. I am sure you will understand...

**Comment by fedelibre** [comment]
> If I use "Music > Save current view settings as default", as advised by Wilbert in his initial comment, then after closing and restarting Frescobaldi, I get a code editor and a music view with the same width as I had previously.

I don't. Perhaps there's some needed setting in the preferences to make it work?
I have *Start with no session* in the Preferences... Don't know if it matters.

**Comment by papioara** [comment]
I tried that one as well, but it did not have the desirede effect either. I think that in the meantime I have gone through and tried all possible options and settings available. The fact is that at least on my system Frescobaldi 'wants' to keep the same width of the Music View part of its window. I even have taken a ruler for verification. It does not matter what size your window is - small, full-size of full-screen: whatever you set and whatever you change manually, it always keeps the same width.

So I would rather call this some kind of a bug and not some overlooked or ignored setting. It forces the user who wants to work full-screen to manually adjust the Music View every time Frescobaldi is fired up. I have also seen this behaviour in version 3.1.3 on MX Linux - so it may have come with version 3 maybe, because I know for sure that it has not been like this in earlier versions.

**Comment by ksnortum** [comment]
On Fri, Mar 3, 2023 at 9:35 AM Jean Abou-Samra ***@***.***>
wrote:

> It's not a "on / off" state, it's a one-time action.
>

That makes sense now.  Thanks.

**Comment by papioara** [comment]
My Frescobaldi config has this in [musicview]:

````
[musicview]

continuousMode=true
document_properties=false
editinplace\dialog\size=@Size(361 195)
kinetic_scrolling=true
magnifier\scale=300
magnifier\size=300
newer_files_only=true
orientation=2
pageLayoutMode=single
rotation=0
show_scrollbars=true
sync_cursor=true
viewMode=0
zoomFactor=0.5
````

Would it be possible to add an entry manually if one knew how to put it for Frescobaldi?

**Comment by papioara** [comment]
Yes, I guess so, but I have no idea about the proper syntax. Many entries are pretty self-explaining, but what for instance would viewMode=0 mean? I mean, of course I could play around with values. Could someone tell me what the syntax is called? Is it Python? If I knew a name, I could maybe try and learn to understand what Frescobaldi might expect as instruction.

**Comment by fedelibre** [comment]
According to [this](https://github.com/frescobaldi/frescobaldi/blob/master/frescobaldi_app/musicview/__init__.py#L139), the setting is supposed to be saved within the `musicview` section of the configuration file.

My frescobaldi.conf has this:

```
[musicview]
continuousMode=true
orientation=2
pageLayoutMode=single
rotation=0
viewMode=0
zoomFactor=1
```

No information about pane sizes.

**Comment by papioara** [comment]
I have looked into all your proposals and I offer you a warm thank you for your assistance and effort, but I have decided to give up on it for now - I lack too much background to really be able to find a solution without investing quite a bit of time. 

Hoping that in a future release this will be fixed/added I will continue to manually adjust the ratio in full-screen mode. After all this is a minor thing. Do you think that it is worth a feature request?
